src/main/java/org/terasology/books/logic/BookcaseSystem.java

```diff
@@ -60,7 +60,7 @@ public void filterBook(BeforeItemPutInInventory event, EntityRef entity) {
     @ReceiveEvent(components = {BookcaseComponent.class})
     public void onDestroyBookCase(CreateBlockDropsEvent event, EntityRef entity) {
         if (entity.hasComponent(InventoryComponent.class)) {
-            Vector3f location = entity.getComponent(LocationComponent.class).getWorldPosition();
+            Vector3f location = entity.getComponent(LocationComponent.class).getWorldPosition(new Vector3f());
```

Comment:
Following the boy scout rule, we should probably leave this in a better state than before. Which means:

- This event handler only does something if the entity has the `InventoryComponent`. We should just add this component to the argument list as `InventoryComponent inventory` and get rid of the `if`-condition.
- We blindly assume the entity also has a `LocationComponent` without any null checks. Since we need the location component as well, that should also be added to the handler args as `LocationComponent location`
- this line would become
	```suggestion
	              Vector3f pos = location.getWorldPosition(new Vector3f());
    ```

Reply:
I updated the event handler. so this should be better.